### ISSUE TITLE
What triggers Math.random() function to refresh?

### ISSUE THREAD

**Issue by audreychenjj-2019** [susanBuck/e28-fall19#19]
Hi, I have a question with regards to the Math.random() function in Javascript. What triggers it to refresh? Clicking any clickable radio buttons or only refreshing the web page? I'm asking about this because I'm in the middle of designing a game in which I use Math.random() function and radio buttons. 

I want to know that in the common high low random integer guessing game how does the Math.random() get trigger? If a viewer guesses 5, the page gives a feedback saying "Too high". Then the viewer lows his guess to 3. Then the page gives a feedback saying "Too low". Do we know for sure 4 is the correct answer? Or we don't know because the first feedback given is no longer valid as the "winning" number is refresh the moment the viewer submits one answer.

**Comment by rokmesojedec** [comment]
Calling the Math.random() function causes it to return a value. Every time you call Math.random() it generates a new random float value (a real number) between 0 and 1.

For the needs of your game it would suffice to call the Math.random() function only once and store it in a variable, like this:

```Javascript
let myRandomNumber = Math.random();
```

You may want to read the [MDN article on Math.random()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random), it explains how to use Math.random() function to generate and integer in an arbitrary interval.

**Comment by susanBuck** [comment]
>> What triggers it to refresh? Clicking any clickable radio buttons or only refreshing the web page?

It's up to you how you program it. It could be triggered as a result of a button being clicked, a radio option being changed, or a page refresh.

In my "Heads or Tails" example I wanted it to be triggered when the "Flip" button was clicked, so I added an event listener to that button, that triggers the function I created called `flip`:

```js
flipBtn.addEventListener('click', flip);
```

In that function, `Math.random()` is invoked:

```js
function flip() {

  [...code redacted for brevity... ]

  let random = Math.floor(Math.random() * 2);

  [...code redacted for brevity... ]
}
```

So, in summary, whenever the button is clicked, a new random number is generated.

In the case of the High/Low number guessing game, you would *not* want a new random number to be generated as the point is to generate one random number and then have the user guess that number. 

Because of this, I would not incorporate any sort of page refresh because you'd lose track of your number (unless you stored it in the browser, but that's unnecessarily complicated for this task).

Here's a rough sketch of how this game might work:

1. On initial page load, generate the random number
2. The page has a text input for guesses and a button to "check" the guess
3. When the button is clicked, it compares the guess against the random number
4. The results ("High", "Low", "Correct") are injected into the page.
